src/merkle_tree/mod.rs

```diff
@@ -0,0 +1,61 @@
+use crate::{hash, hash_pair};
+
+pub struct MerkleTree {
+    hashes: Vec<Vec<u8>>,
+    count: usize
+}
+
+impl MerkleTree {
+    pub fn new(elements: Vec<String>) -> MerkleTree{
+        let leaves = Self::hash_elements(elements.clone());
+        let hashes = Self::build_hashes(leaves);
+        MerkleTree{hashes, count: elements.len()}
+    }
+
+    pub fn root_hash(tree: MerkleTree) -> Vec<u8> {
+        tree.hashes[0].clone()
+    }
+
+    pub fn add(tree: MerkleTree, elements: Vec<String>) -> MerkleTree {
+        let new_leaves = Self::hash_elements(elements);
+        let old_leaves = tree.hashes[{tree.count-1}..].to_vec();
+        let leaves = [old_leaves, new_leaves].concat();
+        let hashes = Self::build_hashes(leaves.clone());
+        MerkleTree{hashes, count: leaves.len()}
+    }
```

Comment:
as mention by edgar. Because you are in a `impl` block you may pass `self` instead of `MerkleTree` because you are implementing this functions for the struct itself.

the code would look like 
```suggestion
    pub fn add(&mut self, elements: Vec<String>) {
        let new_leaves = self.hash_elements(elements);
        let old_leaves = self.hashes[{tree.count-1}..].to_vec();
        let leaves = [old_leaves, new_leaves].concat();
        let count = leaves.len();
        self.hashes = self.build_hashes(leaves).
        self.count = count;
    }
```
thus you avoid calling `Self::` and to return a new merkle tree (unless it is your intention). Also you can store `leaves.len()` in a variable since it is an integer value and implements copy, in that way you can pass leaves to `build_hashes` without cloning it (which is an expensive operation), without the `.clone()` you are loosing `leaves` ownership but it doesn't matter because it is no longer used.